src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -41,6 +42,16 @@ namespace ripple {
 NotTEC
 preflight0(PreflightContext const& ctx)
 {
+    if (ctx.tx.isFlag(tfInnerBatchTxn) && !ctx.rules.enabled(featureBatch))
+        return temINVALID_FLAG;
+
```

Comment:
should we also add an assertion that `tfInnerBatchTxn` must not be set if `PreflightContext` doesnt have a `batchId`?